### PR TITLE
Use direction to check if it's an outgoing leg

### PR DESCRIPTION
**Pull request by monicasarbu**
Reported here: https://discuss.elastic.co/t/ignore-outgoing-doesnt-work-for-me/37201/2

### CODE REVIEW

**Comment by monicasarbu** [comment]
I didn't test the current fix. We need to add tests with "direction" and "ignore_outgoing".

File: libbeat/publisher/preprocess.go [hunk 1]
```diff
@@ -158,14 +158,13 @@ func updateEventAddresses(publisher *PublisherType, event common.MapStr) bool {
...
 		}
-	}
 
-	if publisher.IgnoreOutgoing && dstServer != "" &&
-		dstServer != publisher.name {
-		// duplicated transaction -> ignore it
-		debug("Ignore duplicated transaction on %s: %s -> %s",
-			publisher.name, srcServer, dstServer)
-		return false
+		if publisher.IgnoreOutgoing && event["direction"] == "out" {
```
Comment (urso) [inline libbeat/publisher/preprocess.go]: move the check for IgnoreOutgoing into the else branch line 159. Then we don't have to query the value from hashmap and have it more localized (plus much cheaper execution).

Totally agree with logic of 'direction' and 'ignore_outgoing' implemented by this changeset.

**Comment by tsg** [comment]
Just a note with what we discussed in the team meeting: the current logic for direction "out" is wrong because it's not simply as an else for "in", without checking that the source IP address is actually local. It would make sense I think to fix that as part of this PR, but your call if you prefer to have another PR for that.

**Comment by monicasarbu** [comment]
@tsg @urso I just noticed your comments, after I committed my version. I hope I answered your concerns.

File: libbeat/publisher/preprocess.go [hunk 2]
```diff
@@ -133,6 +133,7 @@ func filterEvent(event common.MapStr) error {
 func updateEventAddresses(publisher *PublisherType, event common.MapStr) bool {
 	var srcServer, dstServer string
 	src, ok := event["src"].(*common.Endpoint)
+	fmt.Println(ok)
```
Comment (tsg) [inline libbeat/publisher/preprocess.go]: Please deletes the Printlns.

Comment (monicasarbu) [inline libbeat/publisher/preprocess.go]: I'll do. Thanks.

**Comment by tsg** [comment]
LGTM.

**Comment by monicasarbu** [comment]
Ready

**Pull request merged**